[Python] Port strings concept exercise from C#. (#760)
exercism/v3: languages/python/exercises/concept/strings/.docs/introduction.md

# Introduction

> A `str` typed object in Python is an immutable sequence of Unicode code points.

Reviewer: It might be useful to explain how Python expects users to manipulate strings. Is it though built-in methods, functions? See [this example](https://github.com/exercism/v3/blob/master/languages/csharp/exercises/concept/strings/.docs/introduction.md).

Reviewer 2: This does need clarification ... strings are manipulated through the use of `overloaded-operators`, `string-methods`, and `builtin-functions`, among others.

Author: Added a bit of context here.

> # Introduction

Reviewer: ```suggestion
```

> A `str` typed object in Python is an immutable sequence of Unicode code points. This could include letters, numbers, punctuation, etc. To manipulate strings, Python provides string methods that can transform a string into other types, create new strings based on method arguments, or return information about the string. Python operators are also overloaded to provide the ability to use them with strings, e.g. to add them together with `+`.

Reviewer: > Python operators are also overloaded to provide the ability to use them with strings, e.g. to add them together with `+`.

This sentence subtly introduces overloading. This might be a little bit early in the track. Would a simpler version also suffice? Something like:

```suggestion
A `str` typed object in Python is an immutable sequence of Unicode code points. This could include letters, numbers, punctuation, etc. To manipulate strings, Python provides string methods that can transform a string into other types, create new strings based on method arguments, or return information about the string. Strings can be concatenated with `+`.
```

Or do you also want to have the student know that `==` and such can be used?
